### PR TITLE
Add the D programming language to GitHub's topics

### PR DESCRIPTION
**Pull request by wilzbach**
- [x] I followed the contributing guidelines: https://github.com/github/explore/blob/master/CONTRIBUTING.md

I am:
  - [ ] Suggesting edits to an existing topic or collection
  - [x] Curating a new topic or collection

************CURATING A NEW TOPIC************

- [x] I've formatted my changes as a new folder directory, named for the topic as it appears in the URL on GitHub (e.g. `https://github.com/topics/[NAME]`)
- [x] My folder contains a `*.png` image (if applicable) and `index.md`
- [x] All required fields in my `index.md` conform to the Style Guide and API docs: https://github.com/github/explore/tree/master/docs

Please explain why you think this topic should be curated:

D is a general purpose systems and applications programming language. It is a high level language, but retains the ability to write high performance code and interface directly with operating system APIs and with hardware. D is a programming language with an emphasis on powerful metaprogramming facilities on top of a robust, efficient core. Idiomatic D programs are commonly as fast as hand-optimized C code, while being reportedly shorter and easier to write, read, and change. A gradual approach to memory safety allows safe programs to be largely automatically checked, whilst relying on small and fast manually-verified primitives.
D's rich feature set offers many opportunities to a variety of projects, without them being ostensibly related to advanced metaprogramming facilities. High-performance code can be terse and highly optimized without the tedium of repeated coding patterns or hand-written specializations. Highly dynamic code can combine static and dynamic language features in an expressive whole. 

tl;dr: it's a popular language with ~10K repositories on GitHub [1] and industry adoption (see e.g. "Organizations using D)

- https://dlang.org/
- https://dlang.org/blog/
- https://en.wikipedia.org/wiki/D_(programming_language)
- https://github.com/dlang
- https://dlang.org/orgs-using-d.html
- https://twitter.com/d_programming

[1] https://github.com/search?utf8=%E2%9C%93&q=language%3AD&type=Repositories&ref=advsearch&l=D&l=

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
Thanks so much for your first contribution! Without people like you submitting PRs we couldn't run this project. You rock, @wilzbach!
